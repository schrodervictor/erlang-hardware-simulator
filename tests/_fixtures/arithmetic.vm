push constant 20
push constant 22
add
push constant 100
sub
neg
