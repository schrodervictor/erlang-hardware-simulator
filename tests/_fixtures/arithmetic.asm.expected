// push constant 20
@20
D=A
@SP
A=M
M=D
@SP
M=M+1
// push constant 22
@22
D=A
@SP
A=M
M=D
@SP
M=M+1
// add
@SP
AM=M-1
D=M
A=A-1
M=D+M
// push constant 100
@100
D=A
@SP
A=M
M=D
@SP
M=M+1
// sub
@SP
AM=M-1
D=M
A=A-1
M=M-D
// neg
@SP
A=M-1
M=-M
